src/java.base/share/classes/java/util/random/RandomGeneratorFactory.java

```diff
@@ -148,6 +153,22 @@ private static Map<String, Provider<? extends RandomGenerator>> createFactoryMap
                 .filter(p -> !p.type().isInterface())
                 .collect(Collectors.toMap(p -> p.type().getSimpleName(), Function.identity()));
         }
+
+        @SuppressWarnings("removal")
+        private static List<Provider<? extends RandomGenerator>> filterBuiltIn(
+                Map<?, Provider<? extends RandomGenerator>> map) {
+            Predicate<Provider<?>> onlyBuiltIn;
+            if (System.getSecurityManager() != null) {
+                // Don't parse annotation for user classes
+                onlyBuiltIn = p -> VM.isSystemDomainLoader(p.type().getClassLoader());
+            } else {
+                onlyBuiltIn = p -> true;
+            }
+            return map.values().stream().filter(onlyBuiltIn)
+                    .filter(p -> !p.type().isAnnotationPresent(Deprecated.class) &&
+                            p.type().isAnnotationPresent(RandomGeneratorProperties.class))
+                    .toList();
```

Comment:
Hello Johannes, I haven't yet fully grasped this change, so a few questions:

1. Does this change now mean that in the presence of a SecurityManager, the `RandomGeneratorFactory.all()` method will no longer return instances of `RandomGeneratorFactory` which belong to the application's classloader (those added as Service provider implementations of that interface within the application's jar for example)? Would that then contradict the expectations of the `RandomGeneratorFactory.all()` API and is there any other way to obtain those application specific implementations?
2. I see that this patch now calls `VM.isSystemDomainLoader` in the presence of the SecurityManager. That method implementation internally calls `ClassLoader.getPlatformClassLoader()` which has SecurityManager checks for the `RuntimePermission` for `getClassLoader` permission. Does this call then have to be in a privileged block?
3. Is this entire issue specific only to the `RandomGeneratorFactory.all()` method? For example, I see that there are some other methods which also use the ServiceProvider API to find the relevant provider implementation and there's also an `isDeprecated` method which checks for the presence of an annotation on the provider's implementation. Are those methods affected too?
4.